velox/functions/lib/DateTimeFormatter.cpp

```diff
@@ -1697,4 +1697,124 @@ std::shared_ptr<DateTimeFormatter> buildJodaDateTimeFormatter(
   return builder.setType(DateTimeFormatterType::JODA).build();
 }
 
+std::shared_ptr<DateTimeFormatter> buildSimpleDateTimeFormatter(
+    const std::string_view& format,
+    bool lenient) {
+  VELOX_USER_CHECK(!format.empty(), "Format pattern should not be empty.");
+
+  DateTimeFormatterBuilder builder(format.size());
+  const char* cur = format.data();
+  const char* end = cur + format.size();
+
+  while (cur < end) {
+    const char* startTokenPtr = cur;
+
+    // Literal case.
+    if (*startTokenPtr == '\'') {
+      // 2 consecutive single quote.
```

Comment:
nit: perhaps clarify what will be done if there are '2 consecutive single quote'.